### PR TITLE
add native return types

### PR DESCRIPTION
**Pull request by orklah**
this PR propose adding quite a lot of missing return types.

They were suggested by phpstorm and then refined with psalm running and analysing itself.

If this goes well, I plan to do the same with parameters.
EDIT: I may also propose a PR to get rid of all the redundant phpdoc.

### CODE REVIEW

**Comment by staabm** [comment]
You might automate this refactorings with rector

**Comment by orklah** [comment]
> You might automate this refactorings with rector

Probably :) I could have done it with psalter too with a tweak or two, but I like how integrated the inspections are in phpstorm. For example, I could choose not to apply types like TCallable(a template in phpdoc) instead of being done and reverted afterwards

**Comment by staabm** [comment]
Sorry I wasn‘t clear: my suggestion was regarding the future possible Patches you mentioned (nit sure those are supported by phpstorm/psalter)

> If this goes well, I plan to do the same with parameters.
EDIT: I may also propose a PR to get rid of all the redundant phpdoc.

**Comment by orklah** [comment]
> Sorry I wasn‘t clear: my suggestion was regarding the future possible Patches you mentioned (not sure those are supported by phpstorm/psalter)
> 
> > If this goes well, I plan to do the same with parameters.
> > EDIT: I may also propose a PR to get rid of all the redundant phpdoc.

Oh yeah, I'll try to see which tool is the most effective if I have time! FYI, both native parameter types and redundant phpdoc are supported by phpstorm on last version
For Psalter, it's kind of a hack, The tool won't suggest to add native types if they're already in phpdoc in a lot of cases. However, you can blind it by disabling phpdoc in the config. Psalter will then diligently put every type in signature if possible. The drawback is when there i a more precise type in phpdoc, it will replace it (given it no longer see it). So when I do that, I patch Psalter so it can't modify phpdoc either

**Comment by muglug** [comment]
Do you know whether or not this affect performance?

**Comment by weirdan** [comment]
AFAIK for non-object types the effect is negligible: https://3v4l.org/PAW5G/perf#output cf https://3v4l.org/plbip/perf#output

**Comment by muglug** [comment]
Please remove the unnecessary docblocks as well – at least, a good chunk of them – I'd prefer a single PR

**Comment by orklah** [comment]
> Do you know whether or not this affects performance?

I remember watching a conference by @nikic which showed that in some case it add additional type checks but in other (rarer) cases it will actually avoid some checks or use specialized opcode instructions and be more effective:
https://youtu.be/JBWgvUrb-q8?t=1588

> Please remove the unnecessary docblocks as well – at least, a good chunk of them – I'd prefer a single PR

Will do!

File: src/Psalm/FileSource.php [hunk 1]
```diff
@@ -26,5 +26,5 @@ public function getRootFilePath();
     /**
      * @return Aliases
      */
-    public function getAliases();
+    public function getAliases(): Aliases;
```
[the remark this answers is not in the document]
Comment (orklah) [reply src/Psalm/FileSource.php:20]: Yeah, to be honest, my goal was to cover a large quantity quickly. If I saw an issue with hierarchy or phpdoc or with an issue reported on psalm, I reverted the type. Maybe that's what happened here.

**Comment by orklah** [comment]
This should be it if the remaining CI pass.

**Review by weirdan — CHANGES_REQUESTED** [review]

**Comment by weirdan** [comment]
Note that it is a BC break to introduce a return type in a non-internal non-final class or interface. So this is a good candidate for 4.0 release.

File: src/Psalm/Internal/ExecutionEnvironment/BuildInfoCollector.php [hunk 2]
```diff
@@ -60,7 +60,6 @@ public function collect() : array
      *
      * "TRAVIS", "TRAVIS_JOB_ID" must be set.
      *
-     * @return $this
```
[the remark this answers is not in the document]
Comment (weirdan) [reply src/Psalm/Internal/ExecutionEnvironment/BuildInfoCollector.php:63]: @orklah there are several more of these changes below

File: src/Psalm/Internal/Analyzer/AlgebraAnalyzer.php [hunk 3]
```diff
@@ -28,7 +28,6 @@ class AlgebraAnalyzer
      * @param  array<int, Clause>   $formula1
      * @param  array<int, Clause>   $formula2
      * @param  StatementsAnalyzer    $statements_analyzer,
```
[the remark this answers is not in the document]
Comment (orklah) [reply src/Psalm/Internal/Analyzer/AlgebraAnalyzer.php]: oh nicely spotted. The comma was considered by phpstorm as a comment in itself

File: src/Psalm/Internal/ExecutionEnvironment/BuildInfoCollector.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (orklah) [reply src/Psalm/Internal/ExecutionEnvironment/BuildInfoCollector.php:63]: fixed as well, thanks

**Comment by muglug** [comment]
Thanks!

**Pull request merged**